web/concrete/src/Permission/Key/Key.php

```diff
@@ -502,4 +502,10 @@ public static function exportTranslations()
 
         return $translations;
     }
+
+    public static function clearCache()
+    {
+        $cache = Core::make('cache/expensive');
```

Comment:
@Remo While I agree with that sentiment, we should consider doing this in another layer and not in the `Key` value object. Adding more static methods and more responsibilities to the Key and other value objects is just going to make it more difficult to migrate in the future.

Static methods on these classes are no different than having global functions, for example there's functionally no difference between calling `Key::add($derp)` and calling something PHP4 procedural like `concrete5_add_permission_key($derp)`, so lets change these things to match our object oriented core and not rely on global functionality. Instead of `Key::add()`, why don't we make a `KeyRepository` that has manages some CRUD operations, that way we can replace `Key::add()` with something like: 

```php

class Key
{

    public static function add()
    {
        if (!$repository = static::getKeyRepository()) {
            throw new \RuntimeException("concrete5 application has not started up yet, this deprecated method requires concrete5's runtime to work.");
        }

        return $repository->add(...);
    }

}
```

That way this repository class can have a single responsibility (Being a repository) and can be aware of whatever it needs to be aware of without adding new stuff to the `Key` class that we need to support to infinitum.

**I don't think this is a complete solution, we must discuss this further before we act on it. This should be part of a larger change in core design and should be a pattern we try to adhere to moving forward.**